datasetviewer/tests/file_loader_presenter_test.py

```diff
@@ -15,58 +15,63 @@ class FileLoaderPresenterTest(unittest.TestCase):
 
     def setUp(self):
 
+        # Create a mock of the MainViewPresenter
         self.mock_main_presenter = mock.create_autospec(MainViewPresenterInterface)
         self.mock_main_presenter.set_dict = mock.MagicMock()
+
+        # Create a mock of the FileLoaderView
         self.mock_view = mock.create_autospec(FileLoaderViewInterface)
 
+        # Create fake empty data, a fake file path, and a mock that returns a fake file path
         self.empty_data = xr.Dataset()
         self.fake_file_path = ("filepath", ".ext")
-
         self.mock_view.get_selected_file_path = mock.MagicMock(side_effect=lambda: self.fake_file_path)
 
-        self.fl_presenter = FileLoaderPresenter(self.mock_view)
-        self.fl_presenter.register_master(self.mock_main_presenter)
-
     def test_presenter_throws_if_view_none(self):
         '''
         Test that the FileLoaderPresenter throws an Exception when the FileLoaderView is None.
         '''
         with self.assertRaises(ValueError):
             FileLoaderPresenter(None)
 
-    def test_register_master(self):
-        '''
-        Test the two-way link between the FileLoaderPresenter and its MainViewPresenter by ensuring that the master's
-        `subscribe_subpresenter` method is called after the FileLoaderPresenter's `register_master` method is called.
-        '''
-        fl_presenter = FileLoaderPresenter(self.mock_view)
-
-        main_presenter = mock.create_autospec(MainViewPresenterInterface)
-        fl_presenter.register_master(main_presenter)
-
-        # main_presenter.subscribe_file_loader_presenter.assert_called_once_with(fl_presenter)
-
     def test_file_selection_calls_get_path_on_view(self):
         '''
-        Test that a FILELOADREQUEST in the FileLoaderPresenter's `notify` function causes the `get_selected_file_path`
+        Test that a FILEOPENREQUEST in the FileLoaderPresenter's `notify` function causes the `get_selected_file_path`
         method on the FileLoaderVIew to be called.
         '''
         fl_presenter = FileLoaderPresenter(self.mock_view)
         fl_presenter.register_master(self.mock_main_presenter)
+
+        # Mock the `_load_data` method so that no actual file-reading occurs
         fl_presenter._load_data = mock.MagicMock()
```

Comment:
Excellent work on the comments. Very useful indeed. 

As a side-point, and maybe something to consider for future work is not to mock methods on anything concrete. Personally I would put the required abstraction in place if you ever find you need to Mock without `create_autospec`